wallet/src/de/schildbach/wallet/ui/dashpay/ContactsFragment.kt

```diff
@@ -108,116 +106,122 @@ class ContactsFragment : BottomNavFragment(R.layout.fragment_contacts_root), Tex
                     "ContactSearchResultsAdapter.Listener")
         }
 
-        contacts_rv.layoutManager = LinearLayoutManager(requireContext())
-        contacts_rv.adapter = this.contactsAdapter
-        this.contactsAdapter.itemClickListener = this
+        binding.contactList.apply {
+            contactsRv.layoutManager = LinearLayoutManager(requireContext())
+            contactsRv.adapter = contactsAdapter
+            contactsAdapter.itemClickListener = this@ContactsFragment
 
-        initViewModel()
+            initViewModel()
 
-        when (mode) {
-            MODE_VIEW_REQUESTS -> {
-                search.visibility = View.GONE
-                icon.visibility = View.GONE
-                setupActionBarWithTitle(R.string.contact_requests_title)
-            }
-            MODE_SEARCH_CONTACTS -> {
-                // search should be available for all other modes
-                search.addTextChangedListener(this)
-                search.visibility = View.VISIBLE
-                icon.visibility = View.VISIBLE
-                setupActionBarWithTitle(R.string.contacts_title)
-            }
-            MODE_SELECT_CONTACT -> {
-                search.addTextChangedListener(this)
-                search.visibility = View.VISIBLE
-                icon.visibility = View.VISIBLE
-                setupActionBarWithTitle(R.string.contacts_send_to_contact_title)
-                forceHideBottomNav = true
+            when (mode) {
+                MODE_VIEW_REQUESTS -> {
+                    search.visibility = View.GONE
+                    icon.visibility = View.GONE
+                    setupActionBarWithTitle(R.string.contact_requests_title)
+                }
+                MODE_SEARCH_CONTACTS -> {
+                    // search should be available for all other modes
+                    search.addTextChangedListener(this@ContactsFragment)
+                    search.visibility = View.VISIBLE
+                    icon.visibility = View.VISIBLE
+                    setupActionBarWithTitle(R.string.contacts_title)
+                }
+                MODE_SELECT_CONTACT -> {
+                    search.addTextChangedListener(this@ContactsFragment)
+                    search.visibility = View.VISIBLE
+                    icon.visibility = View.VISIBLE
+                    setupActionBarWithTitle(R.string.contacts_send_to_contact_title)
+                    forceHideBottomNav = true
+                }
             }
-        }
 
-        search_for_user.setOnClickListener {
-            onSearchUser()
-        }
-
-        search_for_user_suggestions.setOnClickListener {
-            onSearchUser()
-        }
```

Comment:
Removed since `onSearchUser `is already called by `ContactSearchResultsAdapter.Listener`

Reply:
As long as this works.  Some time ago we had one of these `setOnClickListener`s missing and one of the buttons didn't call `onSearchUser`

Reply:
Rechecked again, working fine. There is another way to handle events from the adapter, you declare an action like `var onActionClick: (View) -> Unit = {}`, call it on click of your list element, and simply assign a method to it from the consumer. That way there is no need to deal with interface implementations.

Reply:
ok, then we can do this in a new PR / Jira story.